tests/aws/services/cloudformation/api/test_templates.py

```diff
@@ -0,0 +1,94 @@
+import contextlib
+import os
+import textwrap
+
+import pytest
+from botocore.exceptions import ClientError
+
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid, to_bytes
+
+
+@markers.aws.validated
+@markers.snapshot.skip_snapshot_verify(
+    paths=["$..ResourceIdentifierSummaries..ResourceIdentifiers"]
+)
+def test_get_template_summary(deploy_cfn_template, snapshot, aws_client):
+    snapshot.add_transformer(snapshot.transform.cloudformation_api())
+    snapshot.add_transformer(snapshot.transform.sns_api())
+
+    deployment = deploy_cfn_template(
+        template_path=os.path.join(
+            # This template has no parameters, and so shows the issue
+            os.path.dirname(__file__),
+            "../../../templates/sns_topic_simple.yaml",
+        )
+    )
+
+    res = aws_client.cloudformation.get_template_summary(StackName=deployment.stack_name)
+
+    snapshot.match("template-summary", res)
+
+
+@markers.aws.validated
+@pytest.mark.parametrize("url_style", ["s3_url", "http_path", "http_host", "http_invalid"])
+def test_create_stack_from_s3_template_url(
```

Comment:
Note: This is the new test function, rest of the PR is mostly minor refactoring / relocating..